File: back-end/src/main/java/com/trucks/application/port/OperationRepository.java
package com.trucks.application.port;

import java.util.List;
import java.util.UUID;
import java.util.Optional;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

import com.trucks.domain.entity.Operation;

public interface OperationRepository {

    Operation save(Operation operation);

    Optional<Operation> find(UUID id);

    Page<Operation> findAll(PageRequest pageRequest);

    Operation update(UUID id, Operation operation);

    Operation delete(UUID id);

    List<Operation> deleteAll();
}

File: front-end/src/application/repository/operationRepository.ts
import { Operation } from "../../domain/entity/operation";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";
import { UllUUID } from '../../utils/ull-uuid';

export interface OperationRepository {
    save: (operation: Operation) => Promise<Either<DataError, Operation>>;

    find: (id: UllUUID) => Promise<Either<DataError, Operation>>;
    
    findAll: (pageRequest: any) => Promise<Either<DataError, Operation[]>>;
    
    deleteAll: () => Promise<Either<DataError, Operation[]>>;
}